### PR TITLE
fixed validation of \yii\base\DynamicModel class by adding attribute labels

### PR DESCRIPTION
**Pull request by strorch**
| Q             | A
| ------------- | ---
| Is bugfix?    | ✔️
| New feature?  | ❌
| Breaks BC?    | ❌
| Tests pass?   | ✔️
| Fixed issues  | https://github.com/yiisoft/yii2/issues/18011

### CODE REVIEW

**Review by hiqsol — APPROVED** [review]

**Review by SilverFire — APPROVED** [review]

File: framework/base/DynamicModel.php [hunk 1]
```diff
@@ -237,4 +239,35 @@ public function attributes()
     {
         return array_keys($this->_attributes);
     }
+
+    /**
+     * @param array $labels
+     * @return $this
```
Comment (samdark) [inline framework/base/DynamicModel.php:245]: Need `@since` and method/argument description.

File: framework/base/DynamicModel.php [hunk 2]
```diff
@@ -237,4 +239,35 @@ public function attributes()
...
+     */
+    public function setAttributeLabels(array $labels = [])
+    {
+        $this->_attributeLabels = $labels;
+
+        return $this;
+    }
+
+    /**
+     * @param string $attibute
```
Comment (samdark) [inline framework/base/DynamicModel.php:255]: Need `@since` and method/argument description.

**Review by samdark — COMMENTED** [review]
Need a bit more polishing. Other than that, looks good.

**Comment by samdark** [comment]
Thanks!

**Pull request merged**